core-bundle/src/EventListener/DataContainer/CustomTemplateOptionsCallback.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\Controller;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\DataContainer;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Service\ResetInterface;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * @Callback(table="tl_article", target="fields.customTpl.options")
+ * @Callback(table="tl_content", target="fields.customTpl.options")
+ * @Callback(table="tl_form", target="fields.customTpl.options")
+ * @Callback(table="tl_form_field", target="fields.customTpl.options")
+ * @Callback(table="tl_module", target="fields.customTpl.options")
+ */
+class CustomTemplateOptionsCallback implements ServiceAnnotationInterface, ResetInterface
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var array<string,array<string,string>>
+     */
+    private $fragmentTemplates = [];
+
+    public function __construct(ContaoFramework $framework, RequestStack $requestStack)
+    {
+        $this->framework = $framework;
+        $this->requestStack = $requestStack;
+    }
+
+    public function __invoke(DataContainer $dc): array
+    {
+        /** @var Controller $controllerAdapter */
+        $controllerAdapter = $this->framework->getAdapter(Controller::class);
+
+        switch ($dc->table) {
```

Comment:
What about having individual methods for each dca? This way you would not have to distinguish with a switch statement and it might become more readable.

Reply:
Yeah. By focusing on using invokable listeners I kinda forgot we could do that ;)